src/Symfony/Component/VarDumper/Caster/XmlReaderCaster.php

```diff
@@ -0,0 +1,195 @@
+    {
+        $nodeType = new ConstStub(self::$nodeTypes[$reader->nodeType], $reader->nodeType);
+        $parserProperties = new EnumStub(array(
+            'LOADDTD' => $reader->getParserProperty(\XmlReader::LOADDTD),
+            'DEFAULTATTRS' => $reader->getParserProperty(\XmlReader::DEFAULTATTRS),
+            'VALIDATE' => $reader->getParserProperty(\XmlReader::VALIDATE),
+            'SUBST_ENTITIES' => $reader->getParserProperty(\XmlReader::SUBST_ENTITIES),
+        ));
+
+        if (\XmlReader::NONE === $reader->nodeType) {
+            return $a + self::castXmlNone($nodeType, $parserProperties);
+        }
+
+        if (\XmlReader::ATTRIBUTE === $reader->nodeType) {
+            return $a + self::castAttribute($reader, $nodeType, $parserProperties);
+        }
+
+        if (isset(self::$filteredTypes[$reader->nodeType])) {
+            return $a + self::castFilteredElement($reader, $stub, $nodeType, $parserProperties);
+        }
+
+        $infos = array(
+            'localName' => $reader->localName,
+            'nodeType' => $nodeType,
+
+            'depth' => $reader->depth,
+
+            'attributeCount' => $reader->attributeCount,
+            'hasAttributes' => $reader->hasAttributes,
+
+            'hasValue' => $reader->hasValue,
+            'isDefault' => $reader->isDefault,
+            'isEmptyElement' => $reader->isEmptyElement,
+        );
+
+        if ('' !== $reader->prefix) {
+            $infos['prefix'] = $reader->prefix;
+            $infos['namespaceURI'] = $reader->namespaceURI;
+        }
+
+        if ($reader->hasValue && \XmlReader::TEXT === $reader->nodeType) {
+            $infos['value'] = $reader->value;
+
+            unset($infos['localName']);
+            $stub->cut += 1;
+        }
+
+        if ($reader->hasAttributes) {
+            $infos[Caster::PREFIX_VIRTUAL.'attributes'] = array();
+
+            for ($i = 0; $i < $reader->attributeCount; ++$i) {
+                $infos[Caster::PREFIX_VIRTUAL.'attributes'][] = $reader->getAttributeNo($i);
+            }
+        }
+
+        $infos[Caster::PREFIX_VIRTUAL.'parserProperties'] = $parserProperties;
+
+        return $a + $infos;
+    }
+
+    private static function castXmlNone(ConstStub $type, EnumStub $properties)
+    {
+        return array(
+            'nodeType' => $type,
+            Caster::PREFIX_VIRTUAL.'parserProperties' => $properties,
+        );
+    }
+
+    private static function castFilteredElement(\XmlReader $reader, Stub $stub, ConstStub $type, EnumStub $properties)
+    {
+        // counter for number of elements cut out of the stub
+        $stub->cut += self::BASE_ITEM_NUMBER;
+
+        $infos = array(
+            'localName' => $reader->localName,
+            'nodeType' => $type,
+            'depth' => $reader->depth,
+        );
+
+        $stub->cut -= 3;
+
+        // ignore localName if it is a #text node
+        //
+        // (No significant information brought by this name, the nodeType is
```

Comment:
The point was that having the following :

```
XMLReader {
    localName: "#text"
    nodeType: "TEXT"
}
```

(Or `Whitespace`, ... etc) is redundant, so I chose to ignore the `localName`. But I can readd it...